core/main/src/broker/endpoint_broker.rs

```diff
@@ -619,6 +621,51 @@ impl BrokerOutputForwarder {
                                 if !apply_filter(&broker_request, &result, &rpc_request) {
                                     continue;
                                 }
+
+                                // handle events with internal request if required
+                                if requires_event_handling {
+                                    if let Some(method) = broker_request.rule.event_handler.clone()
+                                    {
+                                        let session_id = rpc_request.ctx.get_id();
+                                        let request_id = rpc_request.ctx.call_id;
+                                        let protocol = rpc_request.ctx.protocol.clone();
+                                        let platform_state_c = platform_state.clone();
+                                        tokio::spawn(async move {
+                                            if let Ok(res) =
+                                                BrokerUtils::handle_main_internal_request(
+                                                    &platform_state_c,
+                                                    method.as_str(),
+                                                )
+                                                .await
+                                            {
+                                                response.result = Some(
+                                                    serde_json::to_value(res.clone()).unwrap(),
+                                                );
+                                            }
+                                            response.id = Some(request_id);
+
+                                            let message = ApiMessage::new(
+                                                protocol,
+                                                serde_json::to_string(&response).unwrap(),
```

Comment:
Depending on the subscription function, this may require transforming the response from the `getter` to the actual `event` response. It should be okay for now, but it is worthwhile to add a TODO comment.